tools/pd-ctl/pdctl/ctl.go

```diff
@@ -86,11 +92,14 @@ func GetRootCmd() *cobra.Command {
 				return err
 			}
 
-			if err := command.InitHTTPSClient(CAPath, certPath, keyPath); err != nil {
+			if err := command.InitHTTPSClient(addrs, CAPath, certPath, keyPath); err != nil {
 				rootCmd.Println(err)
 				return err
 			}
+		} else {
+			command.SetNewPDClient(strings.Split(addrs, ","))
```

Comment:
What about setting the PD HTTP client unconditionally before we finish the replacement?

Reply:
we can simplify it after replacing dialClient